### PR TITLE
Igor/exercise result

### PR DESCRIPTION
**Pull request by igormas**

### CODE REVIEW

File: src/components/exerciseResult/exerciseResultSrv.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            function _getExerciseResultPath(guid) {
+                return 'exerciseResults/' + guid;
+            }
+
+            function _getInitExerciseResult(exerciseTypeId,exerciseId){
+                var storage = InfraConfigSrv.getStorageService();
+                return {
+                    exerciseId: exerciseId,
+                    exerciseTypeId: exerciseTypeId,
+                    startedTime: storage.variables.currTimeStamp,
```
Comment (assafshp) [inline src/components/exerciseResult/exerciseResultSrv.js]: which started time is it ? if it is init function maybe we should

File: src/components/storageSrv/storageSrv.js [hunk 2]
```diff
@@ -53,18 +59,53 @@
...
+                return this.setter(pathStrOrObj, newValue).then(function(res){
+                    if(!angular.isObject(pathStrOrObj)){
+                        var temp = res;
+                        res = {};
+                        res[pathStrOrObj] = temp;
+                    }
+                    var promArr = [];
+                    var retVal = {};
+                    angular.forEach(res,function(val,key){
+                        entityCache.put(key,val);
```
Comment (assafshp) [inline src/components/storageSrv/storageSrv.js]: does it return a promise

File: test/karma-unit.conf.js [hunk 3]
```diff
@@ -46,7 +46,7 @@ module.exports = function (config) {
             'test/mock/**/*.js',
 
             //tests
-            'test/spec/**/*.spec.js'
+            'test/spec/**/*.spec.js'//@todo
```
Comment (assafshp) [inline test/karma-unit.conf.js]: what there is to do here?

File: test/spec/exerciseResult/exerciseResult.spec.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        testStorage.db.examResults = {};
+        testStorage.db.users = {
+            '$$uid':{
+                exerciseResults: {},
+                examResults: {}
+            }
+        };
+    });
+
+    xdescribe('testing get exercise result', function(){
```
Comment (assafshp) [inline test/spec/exerciseResult/exerciseResult.spec.js]: should be in comment?

File: test/spec/exerciseResult/exerciseResult.spec.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+        it('when requesting for a not exiting result then a new initialized result should be returned', function () {
+            var exerciseId = 1;
+            var exerciseResult = actions.getExerciseResult(ExerciseTypeEnum.TUTORIAL.enum, exerciseId);
+            var expectedExerciseResult = {
+                exerciseId: exerciseId,
+                exerciseTypeId: ExerciseTypeEnum.TUTORIAL.enum,
+                startedTime: '%currTimeStamp%',
+                questionResults: []
+            };
+            expect(exerciseResult).toEqual(jasmine.objectContaining(expectedExerciseResult));
```
Comment (assafshp) [inline test/spec/exerciseResult/exerciseResult.spec.js]: nice ! I didn't know that

File: test/spec/storageSrv/storageSrv.spec.js [hunk 6]
```diff
@@ -41,31 +55,78 @@ describe('testing service "StorageSrv":', function () {
         return syncCommunicator;
     };
 
-    it('when getting not exiting data then empty object should be returned', function () {
-        var entityCommunicator = actions.syncEntityCommunicator(new testStorage.EntityCommunicator('test'));
+    it('when getting not exiting data via entity communicator then empty object should be returned', function () {
+        var entityCommunicator = actions.syncEntityCommunicator();
         var val = entityCommunicator.get();
-        expect(val).toEqual({});
+        expect(val).toEqual(jasmine.objectContaining({}));
```
Comment (assafshp) [inline test/spec/storageSrv/storageSrv.spec.js]: in this case, can't it be without objectContaining?

File: src/components/exerciseResult/exerciseResultSrv.js (hunk 1 above)
Comment (igormas) [inline src/components/exerciseResult/exerciseResultSrv.js]: the startedTime will be set by the adapter(storageFirebaseSrv).

File: test/karma-unit.conf.js (hunk 3 above)
Comment (igormas) [inline test/karma-unit.conf.js]: ignore, will be deleted.

File: test/spec/exerciseResult/exerciseResult.spec.js (hunk 4 above)
Comment (igormas) [inline test/spec/exerciseResult/exerciseResult.spec.js]: No, will be removed in the final pull request.

File: test/spec/exerciseResult/exerciseResult.spec.js (hunk 5 above)
Comment (igormas) [inline test/spec/exerciseResult/exerciseResult.spec.js]: Harel left some good stuff in his tests :)

File: test/spec/storageSrv/storageSrv.spec.js (hunk 6 above)
Comment (igormas) [inline test/spec/storageSrv/storageSrv.spec.js]: Fixed

File: src/components/storageSrv/storageSrv.js (hunk 2 above)
Comment (igormas) [inline src/components/storageSrv/storageSrv.js]: No, its sync.

File: src/components/exerciseResult/exerciseResultSrv.js (hunk 1 above)
Comment (assafshp) [inline src/components/exerciseResult/exerciseResultSrv.js]: ok

**Pull request merged**